### ISSUE TITLE
NullPointerException with build 344 in androidx.compose.ui.platform.SkiaLayer

### ISSUE THREAD

**Issue by Syer10** [JetBrains/compose-jb#1157]
I updated my application to build 344 and found a issue with one of my libraries that I ported. The issue does not seem to be on the libraries side as it works fine any build before 344.

### Here is a simple example to reproduce the issue:
Implement `ca.gosyer:accompanist-pager:0.18.1`(or 0.14.0) in gradle

Use this code in the main file
```kotlin
import androidx.compose.ui.window.singleWindowApplication
import com.google.accompanist.pager.ExperimentalPagerApi
import com.google.accompanist.pager.HorizontalPager
import com.google.accompanist.pager.rememberPagerState

@OptIn(ExperimentalPagerApi::class)
fun main() = singleWindowApplication {
    HorizontalPager(rememberPagerState(2)) {}
}
```

Here is the stacktrace that it produces
```kotlin
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at androidx.compose.ui.platform.SkiaLayer.isInLayer-k-4lQ0M(SkiaLayer.desktop.kt:128)
	at androidx.compose.ui.node.LayoutNodeWrapper.withinLayerBounds-3MmeM6k(LayoutNodeWrapper.kt:639)
	at androidx.compose.ui.node.InnerPlaceable.access$withinLayerBounds-3MmeM6k(InnerPlaceable.kt:42)
	at androidx.compose.ui.node.InnerPlaceable.hitTest-9KIMszo(InnerPlaceable.kt:210)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper.hitTestChild-9KIMszo(PointerInputDelegatingWrapper.kt:63)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper.access$hitTestChild-9KIMszo(PointerInputDelegatingWrapper.kt:23)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper$hitTest$1.invoke(PointerInputDelegatingWrapper.kt:51)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper$hitTest$1.invoke(PointerInputDelegatingWrapper.kt:44)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTestInMinimumTouchTarget-N6fFfp4(DelegatingLayoutNodeWrapper.kt:217)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper.hitTest-9KIMszo(PointerInputDelegatingWrapper.kt:44)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper.hitTestChild-9KIMszo(PointerInputDelegatingWrapper.kt:63)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper.access$hitTestChild-9KIMszo(PointerInputDelegatingWrapper.kt:23)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper$hitTest$1.invoke(PointerInputDelegatingWrapper.kt:51)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper$hitTest$1.invoke(PointerInputDelegatingWrapper.kt:44)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTestInMinimumTouchTarget-N6fFfp4(DelegatingLayoutNodeWrapper.kt:217)
	at androidx.compose.ui.node.PointerInputDelegatingWrapper.hitTest-9KIMszo(PointerInputDelegatingWrapper.kt:44)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.LayoutNode.hitTest-9KIMszo$ui(LayoutNode.kt:846)
	at androidx.compose.ui.node.InnerPlaceable.hitTest-9KIMszo(InnerPlaceable.kt:146)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.LayoutNode.hitTest-9KIMszo$ui(LayoutNode.kt:846)
	at androidx.compose.ui.node.LayoutNode.hitTest-9KIMszo$ui$default(LayoutNode.kt:840)
	at androidx.compose.ui.platform.DesktopOwner.onPointerEnter-k-4lQ0M$ui(DesktopOwner.desktop.kt:405)
	at androidx.compose.ui.platform.DesktopOwners.onMouseEntered(DesktopOwners.desktop.kt:207)
	at androidx.compose.ui.awt.ComposeLayer$initCanvas$2$mouseEntered$1.invoke(ComposeLayer.desktop.kt:191)
	at androidx.compose.ui.awt.ComposeLayer$initCanvas$2$mouseEntered$1.invoke(ComposeLayer.desktop.kt:190)
	at androidx.compose.ui.awt.AWTDebounceEventQueue$job$1.invokeSuspend(AWTDebounceEventQueue.desktop.kt:59)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
```

**Comment by Syer10** [comment]
I decided to look at the issue a bit, it seems that [OutlineCache.outline](https://github.com/JetBrains/androidx/blob/desktop/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/platform/SkiaLayer.desktop.kt#L128) is null, which only happens if [OutlineCache.size](https://github.com/JetBrains/androidx/blob/desktop/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/platform/OutlineCache.desktop.kt#L73) is IntSize.Zero. Based on that I believe that a early return of `false` can be done before [this line](https://github.com/JetBrains/androidx/blob/desktop/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/platform/SkiaLayer.desktop.kt#L128) if OutlineCache.outline is null. 

These are just my guesses though, since I don't know the internals of the framework that well.

**Comment by chris-hatton** [comment]
I'm also consistency reproducing this issue in my layout 😞  using build `348`.

**Comment by joreilly** [comment]
Likewise am seeing this pretty consistently in a few projects.

**Comment by igordmn** [comment]
Yes, should be fixed in 1.0.0-beta5

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5276">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
